extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/RequestFailer.java

```diff
@@ -1,78 +0,0 @@
-package io.quarkus.resteasy.runtime;
-
-import java.lang.reflect.Method;
-
-import javax.ws.rs.container.ContainerRequestContext;
-import javax.ws.rs.core.Response;
-
-import org.jboss.resteasy.core.ResteasyContext;
-
-import io.quarkus.security.identity.CurrentIdentityAssociation;
-import io.quarkus.security.identity.SecurityIdentity;
-import io.quarkus.vertx.http.runtime.security.HttpAuthenticator;
-import io.vertx.ext.web.RoutingContext;
-
-/**
- * Helper class for JAXRS request failure responses.
- */
-public class RequestFailer {
-
-    //Servlet API may not be present
-    private static final Class<?> HTTP_SERVLET_REQUEST;
-    private static final Class<?> HTTP_SERVLET_RESPONSE;
-    private static final Method AUTHENTICATE;
-
-    static {
-        Class<?> httpServletReq = null;
-        Class<?> httpServletResp = null;
-        Method auth = null;
-        try {
-            httpServletReq = Class.forName("javax.servlet.http.HttpServletRequest");
-            httpServletResp = Class.forName("javax.servlet.http.HttpServletResponse");
-            auth = httpServletReq.getMethod("authenticate", httpServletResp);
-        } catch (Exception e) {
-
-        }
-        AUTHENTICATE = auth;
-        HTTP_SERVLET_REQUEST = httpServletReq;
-        HTTP_SERVLET_RESPONSE = httpServletResp;
-    }
-
-    public static void fail(ContainerRequestContext requestContext) {
-        if (requestContext.getSecurityContext().getUserPrincipal() == null) {
```

Comment:
@stuartwdouglas what was the case in which this was needed? I moved the logic to https://github.com/quarkusio/quarkus/pull/4413/files#diff-cd55a2a1e7a26ebe9e639f503a059ef1R26 and it is working in the tests without it but I'd like to be sure I'm not missing some scenario in which this is still needed.